network/ipv6Transport.js

```diff
@@ -0,0 +1,263 @@
+         */
+        static async resolveName(name) {
+            try {
+                const dnsResolveDelegate = util.promisify(dns.resolve);
+                return await dnsResolveDelegate(name, 'ANY');
+            } catch (err) {
+                debug(`dnsName: ${name}; error: ${err}`);
+                return null;
+            }
+        }
+
+        _portMapping() {
+            if (!this._upnpClient) {
+                this._upnpClient = natUpnp.createClient();
+            }
+            this._upnpClient.portMapping({
+                public: this._port,
+                private: this._port,
+                ttl: 0
+            }, function (err) {
+                if (err) {debug(`portMapping error`, err);}
+            });
+        }
+
+        /**
+         * For tests
+         */
+        portUnmapping() {
+            if (this._upnpClient) {
+                this._upnpClient.portUnmapping({
+                    public: this._port
+                });
+            }
+        }
+
+        /**
+         * @param {String} address - IP address
+         * @return {Promise<Ipv6Connection>} new connection
+         */
+        connect(address, port) {
+            return new Promise((resolve, reject) => {
+                const socket = net.createConnection(port, address,
+                    async (err) => {
+                        if (err) return reject(err);
+                        resolve(new Ipv6Connection({socket, timeout: this._timeout}));
+                    }
+                );
+                socket.on('error', err => reject(err));
+            });
+        }
+
+        /**
+         * uPnP port mapping, set result addresses
+         */
+        async _getMappingAddress() {
+            let mappings = await this._getPortMappings();
+            if (!mappings || mappings.length === 0) {
+                this._portMapping();
+                mappings = await this._getPortMappings();
+            }
+            if (mappings && mappings.length > 0) {
+                return {privateAddress: mappings[0].private.host};
+            }
+        }
+
+        _setAddresses({privateAddress, publicAddress}) {
+            this._privateAddress = !this.privateAddress ? privateAddress || undefined : this.privateAddress;
+            this._publicAddress = !this._publicAddress ? publicAddress || undefined : this._publicAddress;
+        }
+
+        /**
+         * 
+         * @return {boolean} exists private & public addresses
+         */
+        _existsAddresses() {
+            return ipaddr.isValid(this._privateAddress) && ipaddr.isValid(this._publicAddress);
+        }
+
+        async _getRealAddress() {
+            if (ipaddr.isValid(this._address))
+                return;
+            let addr;
+            try {
+                addr = await this._getPublicAddress();
+            }
+            catch (err) {
+                debug(`Error determining external IP address: ${err}`);
+            }
+            if (!addr)
```

Comment:
codeStyle: или в несколько строк и с `{}` или если без `{}` то в 1 строку
```suggestion
            if (!addr) return;
```